src/main/java/io/github/gitbucket/markedj/Parser.java

```diff
@@ -1,23 +1,25 @@
 package io.github.gitbucket.markedj;
 
+import io.github.gitbucket.markedj.extension.Extension;
 import io.github.gitbucket.markedj.rule.Rule;
 import io.github.gitbucket.markedj.token.*;
 
 import java.util.Map;
+import java.util.Optional;
 import java.util.Stack;
 
 public class Parser {
 
     protected Options options;
     protected Renderer renderer;
+    protected Map<String, Rule> rules;
 
     public Parser(Options options, Renderer renderer){
         this.options = options;
         this.renderer = renderer;
     }
 
     public String parse(Stack<Token> src, Map<String, Lexer.Link> links){
-        Map<String, Rule> rules;
```

Comment:
No need to move `rules` to fields. Or we can move setting up rules to the constructor if we move `rules` to fields.